### ISSUE TITLE
Propagating `DefaultInfo` loses runfiles

### ISSUE THREAD

**Issue by cushon** [bazelbuild/bazel#9442]
Having a rule return the `DefaultInfo` for a dependency appears to lose runfiles from that dep compared to explicitly creating a new `DefaultInfo` that just copies all of the dependencies data. This seems like a bug.

Context: see discussion in https://github.com/bazelbuild/bazel/issues/9391#issuecomment-532139150, which was worked around in #9398. The following change allowed runfiles for the propagated rule to be handled correctly:

Before:

https://github.com/bazelbuild/bazel/blob/b8e1bb18fae89898b0d28535c6355e67b7d4239e/tools/jdk/java_toolchain_alias.bzl#L52

After:

https://github.com/bazelbuild/bazel/blob/66d099e9760aa49b8d93844bd81e9b8c147225dc/tools/jdk/java_toolchain_alias.bzl#L52-L58

**Comment by cushon** [comment]
@laurentlb @iirina

**Comment by davido** [comment]
I tend to agree, that `runtime[DefaultInfo]` should have just worked, and re-creating it sounds rather like a workaround.

**Comment by fmeum** [comment]
I just hit this issue as well (detailed reproducer at https://gist.github.com/fmeum/97bd7198ece1f2c02bfbad11443331ce). It is a very good way to loose quite a bit of time because it's hard to come up with the idea that this *wouldn't* just work.

@brandjon Would you be able to raise the priority back to at least P3? If so, I would start to investigate the root cause and hope to create a PR.
